benches/bench.rs

```diff
@@ -1,234 +1,323 @@
-                _ => (),
+fn fast_xml_namespaced(c: &mut Criterion) {
+    let mut group = c.benchmark_group("fast_xml_namespaced");
+    group.bench_function("untrimmed", |b| {
+        b.iter(|| {
+            let mut r = Reader::from_reader(SAMPLE);
+            r.check_end_names(false).check_comments(false);
+            let mut count = criterion::black_box(0);
+            let mut buf = Vec::new();
+            let mut ns_buf = Vec::new();
+            loop {
+                match r.read_namespaced_event(&mut buf, &mut ns_buf) {
+                    Ok((_, Event::Start(_))) | Ok((_, Event::Empty(_))) => count += 1,
+                    Ok((_, Event::Eof)) => break,
+                    _ => (),
+                }
+                buf.clear();
             }
-            buf.clear();
-        }
-        assert_eq!(count, 1550);
+            assert_eq!(count, 1550);
+        });
+    });
+
+    group.bench_function("trimmed", |b| {
+        b.iter(|| {
+            let mut r = Reader::from_reader(SAMPLE);
+            r.check_end_names(false)
+                .check_comments(false)
+                .trim_text(true);
+            let mut count = criterion::black_box(0);
+            let mut buf = Vec::new();
+            let mut ns_buf = Vec::new();
+            loop {
+                match r.read_namespaced_event(&mut buf, &mut ns_buf) {
+                    Ok((_, Event::Start(_))) | Ok((_, Event::Empty(_))) => count += 1,
+                    Ok((_, Event::Eof)) => break,
+                    _ => (),
+                }
+                buf.clear();
+            }
+            assert_eq!(count, 1550);
+        });
     });
+    group.finish();
 }
 
-#[bench]
-fn bench_fast_xml_escaped(b: &mut Bencher) {
-    let src: &[u8] = include_bytes!("../tests/sample_rss.xml");
-    b.iter(|| {
-        let mut buf = Vec::new();
-        let mut r = Reader::from_reader(src);
-        r.check_end_names(false).check_comments(false);
-        let mut count = test::black_box(0);
-        let mut nbtxt = test::black_box(0);
-        loop {
-            match r.read_event(&mut buf) {
-                Ok(Event::Start(_)) | Ok(Event::Empty(_)) => count += 1,
-                Ok(Event::Text(ref e)) => nbtxt += e.unescaped().unwrap().len(),
-                Ok(Event::Eof) => break,
-                _ => (),
+fn fast_xml_escaped(c: &mut Criterion) {
+    let mut group = c.benchmark_group("fast_xml_escaped");
+    group.bench_function("untrimmed", |b| {
+        b.iter(|| {
+            let mut buf = Vec::new();
+            let mut r = Reader::from_reader(SAMPLE);
+            r.check_end_names(false).check_comments(false);
+            let mut count = criterion::black_box(0);
+            let mut nbtxt = criterion::black_box(0);
+            loop {
+                match r.read_event(&mut buf) {
+                    Ok(Event::Start(_)) | Ok(Event::Empty(_)) => count += 1,
+                    Ok(Event::Text(ref e)) => nbtxt += e.len(),
```

Comment:
Actually, the reason why benchmark is failed here: you accedently removed `.unescaped().unwrap()` part, so you collect escaped lengths instead of unescaped, as can be assumed from the benchmark name.

That change makes things right again:
```suggestion
                    Ok(Event::Text(ref e)) => nbtxt += e.unescaped().unwrap().len(),
```